app/tests/src/ingestion/test_get_grouped_texts.py

```diff
@@ -81,4 +86,9 @@ def test_concatenate_list_items():
             type=TextType.LIST,
             headings=[Heading(title="Overview", level=1)],
         ),
+        EnrichedText(
+            text="New list item in new section.",
+            type=TextType.LIST_ITEM,
+            headings=[Heading(title="Section 1", level=1)],
+        ),
```

Comment:
Commenting here because it's a good illustration: if there's a single list item in the parsed PDF (i.e., it isn't merged with an adjacent paragraph or list item), do we want the type to be LIST_ITEM or LIST? Currently get_grouped_texts always outputs NARRATIVE_TEXT or LIST, this PR also introduces LIST_ITEM. Not a big deal either way, but worth considering which will make future code simpler (since later code that consumes the output from get_grouped_texts and wants to inspect text.type now has to check for both LIST and LIST_ITEM)